grading/load_config_json.cpp

```diff
@@ -145,160 +197,125 @@ void ArchiveValidatedFiles(nlohmann::json &whole_config) {
   }
 }
 
-void AddDockerConfiguration(nlohmann::json &whole_config) {
-
-  //check if docker is globally enabled.
-  if (!whole_config["autograding_method"].is_string()) {
-    whole_config["autograding_method"] = "jailed_sandbox";
-  }else{
-    assert(whole_config["autograding_method"] == "docker"
-        || whole_config["autograding_method"]  == "jailed_sandbox");
-  }
-
-  //check if there are global container defaults present. If not, add an empty object.
-  if(whole_config["container_options"].is_null()){
-    whole_config["container_options"] = nlohmann::json::object();
-  }
-
-  //Fill in the defaults.
-  if(!whole_config["container_options"]["container_image"].is_string()){
-    whole_config["container_options"]["container_image"] = "submitty/autograding-default:latest";
-  }
-
-  if (!whole_config["container_options"]["use_router"].is_boolean()){
-    whole_config["container_options"]["use_router"] = false;
-  }
-
-  if (!whole_config["container_options"]["single_port_per_container"].is_boolean()){
-    whole_config["container_options"]["single_port_per_container"] = false; // connection
-  }
-
-  if (!whole_config["container_options"]["number_of_ports"].is_number_integer()){
-    whole_config["container_options"]["number_of_ports"] = 1; // connection
-  }
-
-  nlohmann::json::iterator tc = whole_config.find("testcases");
-  assert (tc != whole_config.end());
+void AddDockerConfiguration(nlohmann::json &testcases, nlohmann::json &whole_config) {
 
   int testcase_num = 0;
-  for (typename nlohmann::json::iterator itr = tc->begin(); itr != tc->end(); itr++,testcase_num++)
-  {
-    std::string title = whole_config["testcases"][testcase_num].value("title","BAD_TITLE");
+  for (typename nlohmann::json::iterator itr = testcases.begin(); itr != testcases.end(); itr++,testcase_num++) {
+    std::string title = (*itr).value("title","BAD_TITLE");
 
-    nlohmann::json this_testcase = whole_config["testcases"][testcase_num];
```

Comment:
why did you remove this local variable?  (I think it helps with code readability -- *itr is less clear in this lengthy function)

Do we need to switch it to a reference?